LoRaEngine/LoraKeysManagerFacade/ClearLnsCache.cs

```
@@ -0,0 +1,99 @@
+            }
+
+            await ClearLnsCacheInternalAsync(cancellationToken);
+
+            return new AcceptedResult();
+        }
+
+        internal async Task ClearLnsCacheInternalAsync(CancellationToken cancellationToken)
+        {
+            this.logger.LogInformation("Clearing device cache for all edge and Pub/Sub channel based Network Servers.");
+            var edgeDevices = await this.edgeDeviceGetter.ListEdgeDevicesAsync(cancellationToken);
+            var tasks = edgeDevices.Select(e => InvokeClearViaDirectMethodAsync(e, cancellationToken)).ToArray();
+
+            await PublishClearMessageAsync();
+            await Task.WhenAll(tasks);
+        }
+
+        internal async Task PublishClearMessageAsync()
+        {
+            await this.channelPublisher.PublishAsync(LoraKeysManagerFacadeConstants.ClearCacheMethodName, new LnsRemoteCall(RemoteCallKind.ClearCache, null));
+            this.logger.LogInformation("Cache clear message published on Pub/Sub channel");
+        }
+
+        internal async Task InvokeClearViaDirectMethodAsync(string lnsId, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var res = await this.serviceClient.InvokeDeviceMethodAsync(lnsId,
+                                                                           LoraKeysManagerFacadeConstants.NetworkServerModuleId,
+                                                                           new CloudToDeviceMethod(LoraKeysManagerFacadeConstants.ClearCacheMethodName),
+                                                                           cancellationToken);
+                if (HttpUtilities.IsSuccessStatusCode(res.Status))
+                {
+                    this.logger.LogInformation("Cache cleared for {gatewayID} via direct method", lnsId);
+                }
+
+                this.logger.LogError("Direct method call to {gatewayID} failed with {statusCode}. Response: {response}",
```

Should we throw here as well instead of logging to indicate an internal server error?